cpp/oneapi/dal/algo/basic_statistics/backend/gpu/partial_compute_kernel_dense_dpc.cpp

```diff
@@ -164,42 +167,52 @@ auto init_computation(sycl::queue& q,
-                                                  data,
-                                                  result_max,
-                                                  pr::max<Float>{},
-                                                  pr::identity<Float>{},
-                                                  { reduce_event_min });
-    reduce_event_max.wait_and_throw();
-    auto reduce_event_sums = pr::reduce_by_columns(q,
-                                                   data,
-                                                   result_sums,
-                                                   pr::sum<Float>{},
-                                                   pr::identity<Float>{},
-                                                   { reduce_event_min });
-    reduce_event_sums.wait_and_throw();
-    auto reduce_event_sumssquares = pr::reduce_by_columns(q,
-                                                          data,
-                                                          result_sums2,
-                                                          pr::sum<Float>{},
-                                                          pr::square<Float>{},
-                                                          { reduce_event_min });
-    reduce_event_sumssquares.wait_and_throw();
-
+    const auto res_op = desc.get_result_options();
+    if (res_op.test(result_options::min)) {
+        auto reduce_event_min = pr::reduce_by_columns(q,
+                                                      data,
+                                                      result_min,
+                                                      pr::min<Float>{},
+                                                      pr::identity<Float>{},
+                                                      { nobs_update_event });
+        reduce_event_min.wait_and_throw();
+    }
+    if (res_op.test(result_options::max)) {
+        auto reduce_event_max = pr::reduce_by_columns(q,
+                                                      data,
+                                                      result_max,
+                                                      pr::max<Float>{},
+                                                      pr::identity<Float>{},
+                                                      { nobs_update_event });
+        reduce_event_max.wait_and_throw();
+    }
+    if (res_op.test(result_options::sum) || res_op.test(result_options::sum_squares_centered) ||
+        res_op.test(res_mean_varc)) {
+        auto reduce_event_sums = pr::reduce_by_columns(q,
+                                                       data,
+                                                       result_sums,
+                                                       pr::sum<Float>{},
+                                                       pr::identity<Float>{},
+                                                       { nobs_update_event });
+        reduce_event_sums.wait_and_throw();
```

Comment:
Do we really need to wait?

I believe you can do as following:
1) Create an empty `std::vector<sycl::event>`
2) Push events to this vector in each branch
3) Wait for this vector once